Rubberduck.CodeAnalysis/Inspections/Concrete/ExcessiveInterfaceMembersInspection.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using Rubberduck.CodeAnalysis.Inspections.Abstract;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.SettingsProvider;
+
+//todo:
+//finish implementing settings
+//implement quickfix
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete 
+{
+    /// <summary>
+    /// Identifies class modules that define an interface with an excessive number of public members and reminds users about Interface Segregation Principle.
+    /// </summary>
+    /// <why>
+    /// Interfaces should not be designed to continually grow new members; we should be keeping them small, specific, and specialized.
+    /// </why>
+    /// <example hasResult="false">
+    /// <module name="MyModule" type="Class Module">
+    /// <![CDATA[
+    /// Option Explicit
+    /// '@Interface
+    ///
+    /// Public Sub DoSomething()
+    /// 
+    /// End Sub
+    /// ]]>
+    /// </module>
+    /// </example>
+    /// <example hasResult="true">
+    /// <module name="MyModule" type="Class Module">
+    /// <![CDATA[
+    /// Option Explicit
+    /// '@Interface
+    ///
+    /// Public Sub DoSomething1()
+    /// 
+    /// End Sub
+    /// 
+    /// Public Sub DoSomething2()
+    /// 
+    /// End Sub
+    /// 
+    /// '...
+    /// 
+    /// Public Sub DoSomethingNGreaterThanMaxPublicMemberCount()
+    ///  
+    /// End Sub
+    /// ]]>
+    /// </module>
+    /// </example>
+    /// 
+    internal sealed class ExcessiveInterfaceMembersInspection : DeclarationInspectionBase<int>
+    {
+        private static int PublicMemberLimit;
```

Comment:
The information should not be saved in a (static) field. Instead, use the version of the base class passing the global information around the global information as parameter.